python/cudf/cudf/tests/test_text.py

```diff
@@ -778,3 +778,54 @@ def test_read_text(datadir):
     actual = cudf.read_text(chess_file, delimiter=delimiter)
 
     assert_eq(expected, actual)
+
+
+def test_read_text_byte_range(datadir):
+    chess_file = str(datadir) + "/chess.pgn"
+    delimiter = "1."
+
+    with open(chess_file, "r") as f:
+        data = f.read()
+        content = data.split(delimiter)
+
+    # Since Python split removes the delimiter and read_text does
+    # not we need to add it back to the 'content'
+    expected = cudf.Series(
+        [
+            c + delimiter if i < (len(content) - 1) else c
+            for i, c in enumerate(content)
+        ]
+    )
+
+    byte_range_size = len(data) / 3
+
+    actual_0 = cudf.read_text(
+        chess_file,
+        delimiter=delimiter,
+        byte_range=[byte_range_size * 0, byte_range_size],
+    )
+    actual_1 = cudf.read_text(
+        chess_file,
+        delimiter=delimiter,
+        byte_range=[byte_range_size * 1, byte_range_size],
+    )
+    actual_2 = cudf.read_text(
+        chess_file,
+        delimiter=delimiter,
+        byte_range=[byte_range_size * 2, byte_range_size],
+    )
+
+    actual = cudf.concat([actual_0, actual_1, actual_2], ignore_index=True)
```

Comment:
Apologies if I screwed up spacing here, I tried to write something that black would be happy with but you may still have to rerun linters.
```suggestion
    actual = [
        cudf.read_text(
            chess_file,
            delimiter=delimiter,
            byte_range=[byte_range_size * i, byte_range_size],
        ) for i in range(3)
    ]
    actual = cudf.concat(actual, ignore_index=True)
```

Reply:
this is a test file. I prefer the explicit three calls, even if the math is relatively simple. It's easier to debug later if the results need to be inspected. we don't gain anything by making this a range-based expression.

Reply:
You mostly just reduce surface area for accidentally breaking the test by forgetting to change one of these, but sure it's not a big deal. You could always save the list in a separate variable rather than overwriting it with the concat call so that it's still available for inspection upon failure.